src/wfs/src/main/java/org/geoserver/wfs/json/ComplexGeoJsonWriter.java

```diff
@@ -176,7 +176,7 @@ private Property encodeGeometry(Feature feature) {
             }
             // store the attribute name and geometry value of the current feature
             geometryAttribute = feature.getProperty(geometryType.getName());
-            geometry = (Geometry) geometryAttribute.getValue();
+            geometry = geometryAttribute != null ? (Geometry) geometryAttribute.getValue() : null;
```

Comment:
I see, the geometry can be optional in the schema and we were always expecting something. That say, as per the [GeoJSON specification](https://tools.ietf.org/html/rfc7946), if the feature is unallocated it should be represented as a JSON NULL value, citing the specification relevant part:

> A Feature object has a member with the name "geometry".  The value
> of the geometry member SHALL be either a Geometry object as
> defined above or, in the case that the Feature is unlocated, a
> JSON null value.

That say, is worth mentioning, the a JOSN value is represented like [this](https://www.json.org/json-en.html):

```json
{
   "key":null
}
```

Can you double check @Marco in your test that not onyl the value of the geometry attribute is ``NULL`` but also that the key is indeed present?

Reply:
fixed